src/Controls/tests/TestCases.Shared.Tests/Tests/FlakyTestAttribute.cs

```diff
@@ -1,13 +1,84 @@
 ﻿using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
 
-namespace Microsoft.Maui.TestCases.Tests;
-
-public class FlakyTestAttribute : IgnoreAttribute
+namespace Microsoft.Maui.TestCases.Tests
 {
-	public FlakyTestAttribute() : base(nameof(FlakyTestAttribute))
-	{
-	}
-	public FlakyTestAttribute(string reason) : base(reason)
+	// Custom NUnit attribute to mark a test as flaky, allowing retries (by default 2).
+	// If after the retries the test fails, can ignore it.
+	// Note: This attribute should be used temporarily until the test is changed.
+	internal class FlakyTestAttribute : Attribute, IWrapTestMethod, IWrapSetUpTearDown
```

Comment:
Oooh this is cool! I guess it will take some time to actually identify the flaky tests, or we should just very aggressively find the ones we see failing once now and mark it with this attribute and work from there.